sentence_transformers/cross_encoder/losses/ListMLELoss.py

```diff
@@ -0,0 +1,290 @@
+            list_size: Size of the list
+
+        Returns:
+            Tensor: Weights for each position [batch_size, list_size]
+        """
+        if self.rank_discount_fn is not None:
+            return self.rank_discount_fn(ranks)
+
+        # Default rank discount: 2^(list_size - rank) - 1
+        return torch.pow(2.0, list_size - ranks) - 1.0
+
+
+class ListMLELoss(nn.Module):
+    def __init__(
+        self,
+        model: CrossEncoder,
+        lambda_weight: ListMLELambdaWeight | None = None,
+        activation_fct: nn.Module | None = nn.Identity(),
+        mini_batch_size: int | None = None,
+        respect_input_order: bool = True,
+    ) -> None:
+        """
+        ListMLE loss for learning to rank with position-aware weighting. This loss function implements 
+        the ListMLE ranking algorithm which uses a list-wise approach based on maximum likelihood 
+        estimation of permutations. It maximizes the likelihood of the permutation induced by the 
+        ground truth labels with optional position-aware weighting.
+
+        .. note::
+
+            The number of documents per query can vary between samples with the ``ListMLELoss``.
+
+        Args:
+            model (CrossEncoder): CrossEncoder model to be trained
+            lambda_weight (ListMLELambdaWeight, optional): Weighting scheme to use. When specified,
+                implements Position-Aware ListMLE which applies different weights to different rank 
+                positions. Default is None (standard ListMLE).
+            activation_fct (:class:`~torch.nn.Module`): Activation function applied to the logits before computing the
+                loss. Defaults to :class:`~torch.nn.Identity`.
+            mini_batch_size (int, optional): Number of samples to process in each forward pass. This has a significant
+                impact on the memory consumption and speed of the training process. Three cases are possible:
+
+                - If ``mini_batch_size`` is None, the ``mini_batch_size`` is set to the batch size.
+                - If ``mini_batch_size`` is greater than 0, the batch is split into mini-batches of size ``mini_batch_size``.
+                - If ``mini_batch_size`` is <= 0, the entire batch is processed at once.
+
+                Defaults to None.
+            respect_input_order (bool): Whether to respect the original input order of documents.
+                If True, assumes the input documents are already ordered by relevance (most relevant first).
+                If False, sorts documents by label values. Defaults to True.
+
+        References:
+            - Learning to Rank: From Pairwise Approach to Listwise Approach: https://www.microsoft.com/en-us/research/publication/learning-to-rank-from-pairwise-approach-to-listwise-approach/
+            - Position-Aware ListMLE: A Sequential Learning Process for Ranking: https://proceedings.mlr.press/v30/Lan13.html
```

Comment:
```suggestion
            - Position-Aware ListMLE: A Sequential Learning Process for Ranking: https://auai.org/uai2014/proceedings/individuals/164.pdf
```
The old URL doesn't work for me

Reply:
Adding `Listwise approach to learning to rank: theory and algorithm`: [https://dl.acm.org/doi/abs/10.1145/1390156.1390306](https://dl.acm.org/doi/abs/10.1145/1390156.1390306)
would be good, too. I'll change the reference and the citation below.

Reply:
https://github.com/tomaarsen/sentence-transformers/pull/6/commits/af3c6322a247d6d80acdd083c95452d87b1fe424